src/core/lib/gpr/sync_posix.cc

```diff
@@ -133,23 +110,13 @@ void gpr_cv_init(gpr_cv* cv) {
 #if GPR_LINUX
   GPR_ASSERT(pthread_condattr_setclock(&attr, CLOCK_MONOTONIC) == 0);
 #endif  // GPR_LINUX
-
-#ifdef GRPC_ASAN_ENABLED
-  GPR_ASSERT(pthread_cond_init(&cv->cond_var, &attr) == 0);
-  cv->leak_checker = (int*)gpr_malloc(sizeof(*cv->leak_checker));
-  GPR_ASSERT(cv->leak_checker != nullptr);
-#else
   GPR_ASSERT(pthread_cond_init(cv, &attr) == 0);
-#endif
+  LSAN_REGISTER_ROOT_REGION(cv, 1);
```

Comment:
similarly please use sizeof(*cv) if possible, here and below

Reply:
Reverted.